bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/AuditorLedgerCheckerTest.java

```diff
@@ -313,19 +315,22 @@ public void testReadOnlyBookieShutdown() throws Exception {
         final CountDownLatch underReplicaLatch = registerUrLedgerWatcher(count);
 
         int bkIndex = bs.size() - 1;
+        LOG.debug("Moving bookie {} {} to read only...", bkIndex, bs.get(bkIndex));
         ServerConfiguration bookieConf = bsConfs.get(bkIndex);
         BookieServer bk = bs.get(bkIndex);
         bookieConf.setReadOnlyModeEnabled(true);
         bk.getBookie().doTransitionToReadOnlyMode();
 
         // grace period for publishing the bk-ledger
         LOG.debug("Waiting for Auditor to finish ledger check.");
-        assertFalse("latch should not have completed", underReplicaLatch.await(5, TimeUnit.SECONDS));
+        waitForAuditToComplete();
+        assertFalse("latch should not have completed", underReplicaLatch.await(1, TimeUnit.SECONDS));
 
         String shutdownBookie = shutdownBookie(bkIndex);
 
         // grace period for publishing the bk-ledger
         LOG.debug("Waiting for ledgers to be marked as under replicated");
+        waitForAuditToComplete();
         underReplicaLatch.await(5, TimeUnit.SECONDS);
```

Comment:
Would it be better to add an "assertTrue" on the returned value of this "await" ?

Reply:
For a future if it times out then it throws a TimeoutException.  So in this case we don't need the assert.

Reply:
Sorry I misunderstood the line number.  In this case we expect it to finish early, and we then check the results on the lines just below it.  This is mostly making sure that we wait long enough for the edit to be reflected properly.